libs/dh/admin/feature-user-management/src/lib/drawer/dh-edit-user-modal/dh-edit-user-modal.component.ts

```diff
@@ -34,24 +40,48 @@ import {
     WattButtonModule,
     TranslocoModule,
     WattTabsModule,
+    DhUserRolesComponent,
   ],
   templateUrl: './dh-edit-user-modal.component.html',
   styles: ['.margin-right-auto {margin-right:auto}'],
 })
 export class DhEditUserModalComponent {
+  private readonly store = inject(DhAdminUserRolesStore);
   user: UserOverviewItemDto | null = null;
+  private _updateUserRoles: UpdateUserRoles | null = null;
   @ViewChild('editUserModal') editUserModal!: WattModalComponent;
+  @ViewChild('userRoles') userRoles!: DhUserRolesComponent;
 
   open(user: UserOverviewItemDto | null): void {
     this.user = user;
     this.editUserModal.open();
   }
 
-  save(): void {
-    // TODO: Save user
+  save() {
+    if (this.user === null || this._updateUserRoles === null) {
+      this.editUserModal.close(false);
+      return;
+    }
+    this.store.assignRoles({
+      userId: this.user.id,
+      updateUserRoles: this._updateUserRoles,
+    });
+
+    this.store.getUserRolesView(this.user.id);
+    this.userRoles.resetUpdateUserRoles();
+    this.editUserModal.close(true);
   }
```

Comment:
suggestion: maybe we should wait with closing the modal until the request for updating the roles is done? I've tried to do something like that in #1377 . The "Save" button is set to loading while the request is going, so it can't be clicked multiple times. 

issue: since the component is reused in multiple places, a side effect of this can be seen if the request for updating the roles fails. In this case both the component in the drawer and modal show error state as seen in the gif below.

![issue](https://user-images.githubusercontent.com/1096332/216583422-3526b2b2-dee2-408d-8246-94da8e99dab3.gif)

The error state in the modal can probably be fixed by wrapping the modal with an `ngIf` (in order for the component to be destroyed) and flip a boolean that tracks whether the modal is visible or not. I've done it in #1377 

https://github.com/Energinet-DataHub/greenforce-frontend/pull/1377/files#diff-380b32b8b7b20240359f390ad90cefd07c346ee7027f334d757ea7b114e81636R69-R72

This will result in the component not saving state when opened -> interacted with -> closed -> opened again as it is the case now.

Reply:
Fixed in https://github.com/Energinet-DataHub/greenforce-frontend/pull/1374/commits/295086a7a9e7b5814880aba180eb59947be5ffde